alacritty_terminal/src/tty/windows/conpty.rs

```diff
@@ -230,6 +241,35 @@ pub fn new(config: &Options, window_size: WindowSize) -> Option<Pty> {
     Some(Pty::new(conpty, conout, conin, child_watcher))
 }
 
+fn convert_custom_env(custom_env: &HashMap<String, String>) -> Option<Vec<u16>> {
+    let mut result = Vec::new();
+    if custom_env.is_empty() {
+        return None;
+    }
+
+    let mut all_env_keys = HashSet::new();
+    for (custom_key, custom_value) in custom_env {
+        let custom_key = OsStr::new(custom_key);
+        if all_env_keys.insert(custom_key.to_ascii_uppercase()) {
```

Comment:
Is it required to `uppercase` it? Since on e.g. linux you can mix case.

Reply:
Windows env names are case-insensitive and we're required to deduplicate those before sending to the process, so some, upper or lower, common case is needed for that?
Windows, at least before, used to prefer upper case, e.g. PATH, USERNAME, TEMP.

On Linux, env names are case-sensitive and there's no need to do anything like that, as HashMap<String, String> deduplicates things automatically.